### PR TITLE
feat(core): get hook execution stats

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Support `includeRecentState` query for `GET /hooks` and `GET /hooks/:id` to fetch hook recent state.

RecentState:
```ts
export const hookResponseGuard = Hooks.guard.extend({
  recentState: z
    .object({
      successCount: z.number(),
      requestCount: z.number(),
    })
});

export type HookResponse = z.infer<typeof hookResponseGuard>;
```

The recent state is use in the coming feature:
image

## Testing

UT & IT & Test locally.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [x] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/core/src/routes/hook.ts [hunk 1]
```diff
@@ -16,18 +16,32 @@ const nonemptyUniqueHookEventsGuard = hookEventsGuard
...
+  const {
+    hooks: { attachRecentStateToHooks },
+  } = libraries;
+
   router.get(
     '/hooks',
-    koaGuard({ response: Hooks.guard.array(), status: 200 }),
+    koaGuard({
+      query: z.object({ includeRecentState: z.string().optional() }),
+      response: hookResponseGuard.partial({ recentState: true }).array(),
```
Comment (wangsijie) [inline packages/core/src/routes/hook.ts:34]: How about the name "executionStats"

File: packages/core/src/routes/hook.ts [hunk 2]
```diff
@@ -37,15 +51,24 @@ export default function hookRoutes<T extends AuthedRouter>(
...
+        query: { includeRecentState },
       } = ctx.guard;
 
-      ctx.body = await findHookById(id);
+      const hook = await findHookById(id);
+
+      ctx.body = includeRecentState
+        ? {
+            ...hook,
+            recentState: await getHookRecentStateByHookId(hook.id),
```
Comment (wangsijie) [inline packages/core/src/routes/hook.ts:69]: use `attachRecentStateToHooks`?

File: packages/core/src/routes/hook.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/core/src/routes/hook.ts]: Updated.

File: packages/core/src/routes/hook.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/core/src/routes/hook.ts]: Updated.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**